### DISCUSSION TITLE
copilot needs to take types in account, more.

### DISCUSSION THREAD

**Discussion by phifa** [community/community#35744] [Copilot Conversations]
I know copilot takes interfaces and types into account. However, I am working in a fully typed TypeScript project and I get invalid suggestions for Enums for example.

**Comment by iamcjwilliams** [comment]
@phifa Can you provide more detail / or a screenshot of the error you're receiving?

**Comment by phifa** [comment]
```
enum FileType {
  PDF,
  PNG,
}

interface CustomFile {
  file: FileType;
}


const myFile: CustomFile = {
  file: ... 
}

```

Sometimes the suggestions here `...`, are not only `FileType.PDF` or `FileType.PNG`. Sometimes there are other suggestions, even though the enum type narrows it down to only two possible values. 

So I am not getting an error, just invalid enum suggestions from time to time.

**Comment by iamcjwilliams** [comment]
@phifa Our copilot team is investigating this issue

**Comment by iamcjwilliams** [comment]
@phifa Thank you for your feedback! Our copilot team is working on addressing this in future releases!
